lib/engine/game/g_1849/game.rb

```diff
@@ -282,13 +319,18 @@ def price_movement_chart
         def setup
           setup_companies
           afg # init afg helper
-          remove_corp if @players.size == 3
+          remove_corp if @players.size == 3 || (@players.size == 4 && reduced_4p_corps?)
           @corporations[0].next_to_par = true
 
           @available_par_groups = %i[par]
 
           @player_debts = Hash.new { |h, k| h[k] = 0 }
           @moved_this_turn = []
+
+          # below is for variant rules
+          @e_tokens_enabled = false
+          @buy_tokens_enabled = false
+          @issue_bonds_enabled = false
```

Comment:
These variables do not need to be initialized. Can delete from here.

Reply:
done

Reply:
done